client/http-client/src/client.rs

```diff
@@ -77,10 +82,20 @@ impl HttpClientBuilder {
 		self
 	}
 
+	/// Adds a custom header, which is sent with every request
```

Comment:
can you add some additional docs that explains that "dupllicate" keys will be overwritten by this method?